modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/form_builder/form_builder_dd_support.js

```diff
@@ -0,0 +1,730 @@
+						columnsBreakPoints.item(dropNode.getData('layout-position') + i).setStyle('display', 'block');
+					}
+					else if (!increment && columnsBreakPoints.item(dropNode.getData('layout-position') - i)) {
+						columnsBreakPoints.item(dropNode.getData('layout-position') - i).setStyle('display', 'block');
+					}
+				}
+			},
+
+			updateDragAndDropBySidebar: function(event, dragFieldNode) {
+				var instance = this;
+
+				var appendNode = event.currentTarget.appendNode;
+				var fieldSetId = dragFieldNode.getData('field-set-id');
+				var fieldType = FieldTypes.get(dragFieldNode.attr('data-field-type-name'));
+
+				if (!appendNode) {
+					return;
+				}
+				else if (!appendNode.ancestor('.col') || appendNode.ancestor('.col').one('.' + CSS_FIELD)) {
+					appendNode.remove();
+					return;
+				}
+
+				instance.removeEmptyFormClass();
+
+				instance._newFieldContainer = appendNode.ancestor('.col').getData('layout-col');
+				appendNode.remove();
+
+				if (fieldSetId) {
+					instance._addFieldSetInDragAndDropLayout(fieldSetId);
+				}
+				else {
+					instance.createNewField(fieldType);
+					instance._traverseFormPages();
+					instance._destroySortable(instance.sortable1);
+					instance._applyDragAndDrop();
+				}
+			},
+
+			_addDragAndDropActions: function() {
+				var instance = this;
+
+				instance.sortable1 = new A.SortableLayout(
+					{
+						delegateConfig: {
+							target: false,
+							useShim: false
+						},
+						dragNodes: '.layout-col-content',
+						dropNodes: '.layout-row .col-empty',
+						proxy: null,
+						proxyNode: '<div></div>'
+					}
+				);
+
+				instance.afterDragStart(instance.sortable1);
+
+				instance.afterPlaceholderAlign(instance.sortable1);
+
+				instance.afterDragEnd(instance.sortable1);
+			},
+
+			_addFieldSetInDragAndDropLayout: function(fieldSetId) {
+				var instance = this;
+
+				instance._getFieldSetDefinitionRetriever(
+					fieldSetId,
+					function(fieldSetDefinition) {
+						instance.createFieldSet(fieldSetDefinition);
+						instance.unformatFieldsetRows();
+						instance._destroySortable(instance.sortable1);
+						instance._traverseFormPages();
+						instance._applyDragAndDrop();
+					}
+				);
+			},
+
+			_addToStack: function(node) {
+				var instance = this;
+
+				instance._clearStack();
+
+				node.addClass('col-empty-over');
+				instance.activeDropColStack.push(node);
+			},
+
+			_adjustEmptyForm: function(activeLayout) {
+				var instance = this;
+
+				if (activeLayout.get('rows').length == 1 && instance._verifyEmptyForm(activeLayout.get('rows')[0]) && !A.one('#lfr-initial-col-message')) {
+					var columnMessageNode = A.Node.create('<div/>');
+
+					columnMessageNode.text(Liferay.Language.get('drag-from-sidebar-and-drop-here'));
+					columnMessageNode.setAttribute('id', 'lfr-initial-col-message');
+
+					activeLayout.get('rows')[0].get('cols')[0].get('node').addClass('lfr-initial-col');
```

Comment:
reuse this statement `activeLayout.get('rows')[0].get('cols')[0].get('node')`  and avoid chaining